docs/development/plugin_test_en.md

```diff
@@ -0,0 +1,112 @@
+
+    println "  loading ${it.getPlugin().class.name}..."
+    it.getPlugin().load()
+    println "  ...loaded ${it.getPlugin().class.name}"
+
+    println "  starting ${it.getPlugin().class.name}..."
+    it.getPlugin().start()
+    println "  ...started ${it.getPlugin().class.name}"
+}
+println ""
+println "Total number of plugins: ${plugins.size()}"
+```
+
+> **Warning:** This method only shows some basic information about the installed plugins, but does not detect runtime-issues.
+
+## Testing using PCT (Plugin Compatibility Tester)
+
+Jenkins has an official compatibility tester: (https://github.com/jenkinsci/plugin-compat-tester)[]
+This tester can test all plugins contained in a Jenkins WAR-file. For every plugin (`*.hpi` file) the following steps are executed:
+* extract the `pom.xml` from the `*.hpi`-file
+* checkout the corresponding source-code-repository
+* patch the Jenkins-Version in the pom.xml to use the Jenkins-Version from the WAR-file
+* run the tests via maven (`mvn test`)
+
+> All plugins (`*.hpi` files) must be located inside the `/WEB-INF/plugins` directory inside the WAR-file!
+
+> This process can take a lot of time (depending on the amount of plugins)!
+> Also it is prone to errors. When a test for one plugin fails, the whole process fails. 
+
+To start the tester execute:
+```shell
+java -jar pct.jar test-plugins --war "$(pwd)/jenkins-with-plugins.war" --working-dir "$(pwd)/pct-work"
+```
+> Use absolute paths in the command! 
+
+The `pct.jar` can be created using `mvn clean package` from the pct-repository.
+
+### Problems using PCT
+Some of the plugins are quite old and maybe deprecated. The PCT has some problems testing these plugins
+
+#### Unauthenticated Git protocol
+The `pom.xml` might contain a reference to the source-code-repository using an unauthenticated Git protocol on port 9418 like `git://github.com/jenkinsci/async-http-client-plugin.git`.
+To fix this the `pom.xml` inside the `hpi`-file inside the WAR-file must be edited. The `git://`-protocol can be replaced with `https://`.
+
+#### Blocked maven-mirrors
+Maven blocks downloads from mirrors using `http`. When a plugin uses `http`-mirrors a workaround is to disable the block in the maven `settings.xml`.
+Comment out the mirror block the looks like this:
+```xml
+<!-- commented out to allow http-mirrors
+<mirror>
+    <id>maven-default-http-blocker</id>
+    <mirrorOf>external:http:*</mirrorOf>
+    <name>Pseudo repository to mirror external repositories initially using HTTP.</name>
+    <url>http://0.0.0.0/</url>
+    <blocked>true</blocked>
+</mirror> 
+-->
+```
+
+## Conclusion
+
+As time writing (20.06.2023) both test methods were not very useful / successful, because some of the plugins currently in use, can not be tested.
+Furthermore, both methods could not identify known errors, because one plugin (async-http-client) uses reflection to load some classes at runtime, which are missing in JAVA 17.
```

Comment:
```suggestion
Furthermore, both methods could not identify known errors, because one plugin (async-http-client) uses reflection to load some classes at runtime, which are missing in Java 17.
```